tests/unit/workflow/test_annotation.py

```diff
@@ -32,11 +35,29 @@
 
 
 def test__serialize__simple() -> None:
-    assert BoundingBox(top_left=(1, 1), bottom_right=(2, 2))._to_dict() == {
-        "top_left": [1, 1],
-        "bottom_right": [2, 2],
+    obj = LabeledPolygon(points=[(1, 1), (2, 2), (3, 3)], label="test")
+    obj_dict = obj._to_dict()
+    assert obj_dict == {
+        "label": "test",
+        "points": [[1, 1], [2, 2], [3, 3]],
+        DATA_TYPE_FIELD: f"{_AnnotationType._data_category()}/{_AnnotationType.POLYGON.value}",
+    }
+    assert LabeledPolygon._from_dict(obj_dict) == obj
+
+
+def test__serialize__derived() -> None:
+    obj = BoundingBox(top_left=(0, 0), bottom_right=(0, 0))
+    obj_dict = obj._to_dict()
+    assert obj_dict == {
+        "top_left": [0, 0],
+        "bottom_right": [0, 0],
+        "width": 0,
+        "height": 0,
+        "area": 0,
+        "aspect_ratio": 0,
         DATA_TYPE_FIELD: f"{_AnnotationType._data_category()}/{_AnnotationType.BOUNDING_BOX.value}",
     }
+    assert BoundingBox._from_dict(obj_dict) == obj
```

Comment:
Nit: verified the scenario below works, but slightly different from the test here:

```
def test__deserialize__derived() -> None:
    obj_dict = {
        "top_left": [10, 10],
        "bottom_right": [20, 20],
        DATA_TYPE_FIELD: f"{_AnnotationType._data_category()}/{_AnnotationType.BOUNDING_BOX.value}",
    }
    assert BoundingBox._from_dict(obj_dict) == BoundingBox(top_left=(10, 10), bottom_right=(20, 20))
```

Reply:
Added check for deserialization from dict without derived fields 👍